scrapers/scrapers/spiders/mo-bills.py

```diff
@@ -0,0 +1,687 @@
+    ("Second Read", "reading-2"),
+    # make sure passage is checked before reading-3
+    ("Third Read and Passed", "passage"),
+    ("Reported Do Pass", "committee-passage"),
+    ("Voted Do Pass", "committee-passage"),
+    ("Third Read", "reading-3"),
+    ("Referred", "referral-committee"),
+    ("Withdrawn", "withdrawal"),
+    ("S adopted", "passage"),
+    ("Truly Agreed To and Finally Passed", "passage"),
+    ("Signed by Governor", "executive-signature"),
+    ("Approved by Governor", "executive-signature"),
+    ("Vetoed by Governor", "executive-veto"),
+    ("Vetoed in Part by Governor", "executive-veto-line-item"),
+    ("Legislature voted to override Governor's veto", "veto-override-passage"),
+]
+
+TIMEZONE = pytz.timezone("America/Chicago")
+
+custom_headers = {
+    "user-agent": "openstates.org"
+}
+
+
+class UnrecognizedSessionType(BaseException):
+    def __init__(self, session):
+        super().__init__(
+            f"Session {session} has the unrecognized session types."
+        )
+
+
+def lxmlize(url):
+    response = requests.get(url, headers=custom_headers)
+    return Selector(response)
+
+
+# remove whitespace, linebreaks, and end parentheses
+def clean_text(text):
+    newtext = re.sub(r"[\r\n]+", " ", text)
+    newtext = re.sub(r"\s{2,}", " ", newtext)
+    m = re.match(r"(.*)\(.*?\)", newtext)
+    if not m:
+        return newtext.strip()
+    else:
+        return m.group(1).strip()
+
+
+def house_get_actor_from_action(text):
+    m = re.search(r"\((\bH\b|\bS\b)\)", text)
+    if not m:
+        if text.endswith("Governor"):
+            return Chamber.UPPER
+        else:
+            return Chamber.LOWER
+
+    abbrev = m.group(1)
+    if abbrev == "S":
+        return Chamber.UPPER
+    return Chamber.LOWER
+
+
+def senate_get_actor_from_action(text):
+    if re.search("Prefiled", text):
+        return Chamber.UPPER
+
+    m = re.search(r"(\bH\b|\bS\b|House)", text)
+    if not m:
+        if text.endswith("Governor"):
+            return Chamber.EXECUTIVE
+        else:
+            return Chamber.UPPER
+
+    if m.group(1) == "S":
+        return Chamber.UPPER
+    else:
+        return Chamber.LOWER
+
+
+def get_action(actor, action):
+    # Alright. This covers both chambers and everything else.
+    categories = []
+    for flag, acat in flags:
+        if flag in action:
+            categories.append(acat)
+
+    return categories or None
+
+
+class BillsSpider(Spider):
+    name = "mo-bills"
+    session = None
+    chamber = None
+
+    subjects = defaultdict(list)
+
+    def __init__(self, session=None, chamber=None, **kwargs):
+        self.session = session
+        self.chamber = chamber
+        super().__init__(**kwargs)
+
+    def start_requests(self):
+        self.parse_subjects(self.session)
+
+        if self.chamber in [Chamber.UPPER, None]:
+            # scrape senate bills
+            sb_url = 'https://www.senate.mo.gov/23info/BTS_Web/BillList.aspx?SessionType=R'
```

Comment:
The session value in this URL is hard-coded here. Please add support for using the `session` parameter to construct `sb_url` like the original scraper: https://github.com/openstates/openstates-scrapers/blob/74648a9e6091c7e1f5e70c15572b4278bb2fdd71/scrapers/mo/bills.py#L655

Reply:
@jessemortenson I've fixed this issue.